File: backend/src/TaskMgmt.Domain/Entities/AttendanceRecord.cs
using TaskMgmt.Domain.Common;

namespace TaskMgmt.Domain.Entities;

public class AttendanceRecord : AuditableEntity
{
    public required Guid UserId { get; set; }
    public User? User { get; set; }
    public required DateOnly WorkDate { get; set; }
    public DateTimeOffset? CheckInAtUtc { get; set; }
    public double? CheckInLatitude { get; set; }
    public double? CheckInLongitude { get; set; }
    public Guid? CheckInLocationId { get; set; }
    public Location? CheckInLocation { get; set; }
    public DateTimeOffset? CheckOutAtUtc { get; set; }
    public double? CheckOutLatitude { get; set; }
    public double? CheckOutLongitude { get; set; }
    public Guid? CheckOutLocationId { get; set; }
    public Location? CheckOutLocation { get; set; }
}

File: mobile/taskmgmt_app/lib/features/attendance/domain/entities/attendance_record.dart
class AttendanceRecord {
  const AttendanceRecord({
    required this.id,
    required this.workDate,
    required this.checkInAtUtc,
    required this.checkInLocationName,
    required this.checkOutAtUtc,
    required this.checkOutLocationName,
  });

  final String id;
  final DateTime workDate;
  final DateTime? checkInAtUtc;
  final String? checkInLocationName;
  final DateTime? checkOutAtUtc;
  final String? checkOutLocationName;

  bool get isCheckedIn => checkInAtUtc != null;
  bool get isCheckedOut => checkOutAtUtc != null;
}
